src/graph.cpp

```diff
@@ -190,19 +203,31 @@ bool Graph::inSolution(unsigned int j, const unsigned int *solution, unsigned in
     return false;
 }
 
+/**
+ * Recursive function for the backtracking function
+ * Time Complexity: O(2^N) (worst case)
+ * @param currentSolution - Array of the path taken so far
+ * @param currentSolutionDist - Weight of the path taken so far
+ * @param currentNodeIdx - Id of the node the algorithm is currently on
+ * @param bestSolutionDist - Weight of the best path obtained so far
+ * @param bestSolution - Array of the best path obtained so far
+ * @param n - Number of nodes in the graph
+ * @param dists - Distance Matrix for the graph
+ */
 void
-Graph::tspRecursion(unsigned int *currentSolution, unsigned int currentSolutionDist,
+Graph::tspRecursion(unsigned int *currentSolution, double currentSolutionDist,
```

Comment:
Why have dists as a parameter here when you can just access the graph's distance matrix directly?